### DISCUSSION TITLE
Left-hand and right-hand alphabet in `goto_word`

### DISCUSSION THREAD

**Discussion by HairlessVillager** [helix-editor/helix#11199] [Ideas]
In the current version, there's only one alphabet that `goto_word` labels use (configure see the `jump-label-alphabet` in [`Editor` Section](https://docs.helix-editor.com/master/editor.html#editor-section)), and it's not distinguishing between left and right hands. So it's frequent that left hand hit the keyboard twice but right hand takes no action (or the opposite).

IMO it's always a left hand key first and a right hand key second is a good practice. For example:

- `ae` is ❌ because `a` and `e` is both on the left part of keyboard.
- `ju` is ❌ because `j` and `u` is both on the right part of keyboard.
- `wh` is ⭕ because `w` is on the left and `h` is on the right.

I've studied Rust and read the relative source code. Now I have confidence to implement this feature.🤗

**Comment by nik-rev** [comment]
> In the current version, there's only one alphabet that `goto_word` labels use (configure see the `jump-label-alphabet` in [`Editor` Section](https://docs.helix-editor.com/master/editor.html#editor-section)), and it's not distinguishing between left and right hands. So it's frequent that left hand hit the keyboard twice but right hand takes no action (or the opposite).
> 
> IMO it's always a left hand key first and a right hand key second is a good practice. For example:
> 
>     * `ae` is ❌ because `a` and `e` is both on the left part of keyboard.
> 
>     * `ju` is ❌ because `j` and `u` is both on the right part of keyboard.
> 
>     * `wh` is ⭕ because `w` is on the left and `h` is on the right.
> 
> 
> I've studied Rust and read the relative source code. Now I have confidence to implement this feature.🤗

what about people not using qwerty? Not everyone uses the same keyboard layout. Different languages may place keys in different areas.

**Reply by HairlessVillager** [reply]
You're right! There are indeed different keyboard layouts. But if this feature is implemented, you can set the key positions for the left and right hands separately. Here's an example of a layout different from QWERTY:

![image](https://github.com/user-attachments/assets/19ba53a2-71d1-4ccc-88f8-3bc4ffd02803)

In this example, we can set the key positions for the left hand to be `asht` and for the right hand to be `neoi` (this is just an example, you can actually add as many keys as you want) to make it more convenient for you to type.

You could EVEN set the right hand's keys to `asht` and the left hand's to `neoi`, which is completely feasible according to the configuration rules, although no one would actually do that with this keyboard layout because it would definitely make typing more difficult.
